x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlDailyMaintenanceService.java

```diff
@@ -413,13 +423,40 @@ private void handleRolloverResults(String[] indices, List<Exception> failures, A
         finalListener.onResponse(AcknowledgedResponse.FALSE);
     }
 
-    // public for testing
-    public void triggerRollResultsIndicesIfNecessaryTask(ActionListener<AcknowledgedResponse> finalListener) {
-        logger.info("[ML] maintenance task: triggerRollResultsIndicesIfNecessaryTask");
+    // Helper function to check for the "index.lifecycle.name" setting on an index
+    private boolean hasIlm(String indexName) {
```

Comment:
Consider adding a unit test or a BWC test to verify that this works as expected.